### DISCUSSION TITLE
How does CUE parse value supplied to its "--inject" flag?

### DISCUSSION THREAD

**Discussion by seh** [cuelang/cue#932] [Q&A]
Moving here from [the "general" channel in the "CUE" Slack workspace](https://cuelang.slack.com/archives/CLT3ULF6C/p1619624387178000), I have a CUE file with a `@tag` attribute defined like this:
```cue
built_at: int @tag(builtat,type=int)
```
If I run _cue export file.cue --inject builtat=1_, it works as expected.
If I run _cue export file.cue --inject builtat=x_, it fails with this message:
```
built_at: reference "x" not found:
    builtat:1:1
```
That sort of makes sense, though I’m not sure what it means by “reference” there. With what is CUE trying to match the string “x”?

Also interesting: If I instead run  _cue export file.cue --inject ’builtat={x}’,_ with curly braces around “x,” it fails with the same message, having “unwrapped” the value “{x}“:
```
built_at: reference "x" not found:
    builtat:1:2
```
Is there some syntax for the right-hand side of that equals sign that CUE is parsing? The documentation in cue injection doesn’t mention anything elaborate.

**Comment by myitcv** [comment]
@seh - sincere apologies for being slow at replying to this (and thank you for raising it as a discussion from Slack in the first place).

This looks like a bug to me: injected values can only be concrete. AFAICT, that error comes from the compiler, and if the injection has got that far without being detected as a type error. Hence I've raised: https://github.com/cuelang/cue/issues/1010

But in investigating this I stumbled across another issue that looks like a typo/copy paste bad error message, raised as: https://github.com/cuelang/cue/issues/1011

**Comment by cueckoo** [comment]
This discussion has been migrated to https://github.com/cue-lang/cue/issues/932.

For more details about CUE's migration to a new home, please see https://github.com/cue-lang/cue/issues/1078.
